src/Router.js

```diff
@@ -89,22 +89,23 @@ class Router {
   unmount(){
     removeEventListener('popstate', this.rerender)
   }
-}
 
-const staticResolver = (mapper) => {
-  const router = instantiatePathnameRouter(mapper)
-  return (location) => router.resolve(location.pathname)
-}
+  staticResolver(mapper){
+    const router = this.instantiatePathnameRouter(mapper)
+    return (location) => router.resolve(location.pathname)
+  }
 
-const dynamicResolver = (mapper) => {
-  return (location, props) =>
-    instantiatePathnameRouter(mapper, props).resolve(location.pathname)
-}
+  dynamicResolver(mapper){
+    return (location, props) =>
+      this.instantiatePathnameRouter(mapper, props).resolve(location.pathname)
+  }
 
-const instantiatePathnameRouter = (mapper, props) => {
-  const router = new PathnameRouter
-  const map = (path, Component, params={}) =>
-    router.map(path, {Component, ...params})
-  mapper.call(null, map, props)
-  return router
-}
+  instantiatePathnameRouter(mapper, props){
+    const router = new PathnameRouter
+    const map = (path, Component, params = {}) =>
+      router.map(path, {Component, ...params})
+    mapper.call(null, map, props, this.redirectTo)
```

Comment:
I don't think you want to call `redirectTo` within the router. This is a feature I would like to add but I think it needs to return a valid React Component since this is called within a react re-render cycle.